### PR TITLE
Adding test for marking unhealthy node in remediation process.

### PR DESCRIPTION
**Pull request by smoshiur1237**
This PR will add test for marking unhealthy node in Metal3-dev-env  

Co-authored by: Jan Tilles, jan.tilles@est.tech

### CODE REVIEW

**Comment by jan-est** [comment]
This looks good to me

**Review by Jaakko-Os — COMMENTED** [review]
looks good to me

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml [hunk 1]
```diff
@@ -138,4 +138,83 @@
...
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w provisioning | wc -l
+    retries: 10
+    delay: 20
+    register: provisioning_hosts
+    until: provisioning_hosts.stdout == "0"
+
+  - name: Remove unhealthy annotation from "{{ WORKER_BMH }}"
+    shell: |
+      kubectl annotate bmh "{{ WORKER_BMH }}" -n "{{ NAMESPACE }}" capi.metal3.io/unhealthy-
+    when: provisioning_hosts.stdout == "0"
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:213]: Maybe we should remove the condition here and instead rely on the task before this?

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml [hunk 2]
```diff
@@ -138,4 +138,83 @@
...
+  
+  - pause:
+     minutes: 1
+
+  - name: Wait and verify that none of the nodes will not start provisioning
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w provisioning | wc -l
+    retries: 10
+    delay: 20
+    register: provisioning_hosts
+    until: provisioning_hosts.stdout == "0"
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:208]:
```suggestion
    until: 
      - provisioning_hosts.stdout == "0"
      - provisioned_hosts.stdout == "3"
```

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml [hunk 3]
```diff
@@ -138,4 +138,83 @@
...
+  # Start Unhealthy node testing
+  - name: Get KCP name
+    shell: |
+      kubectl get kcp -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
+      | .metadata.name'
+    register: kcp_name
+
+  - name: Define KCP_NAME variable
+    set_fact:
+      KCP_NAME: "{{ kcp_name.stdout }}"
```
[the remark this answers is not in the document]
Comment (smoshiur1237) [reply vm-setup/roles/v1aX_integration_test/tasks/remediation.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml [hunk 4]
```diff
@@ -138,4 +138,83 @@
...
+
+  - name: Mark "{{ WORKER_BMH }}" as unhealthy
+    shell: |
+      kubectl annotate bmh "{{ WORKER_BMH }}" -n "{{ NAMESPACE }}" capi.metal3.io/unhealthy=
+ 
+  - name: Delete worker Machine object "{{ WORKER_NODE }}"
+    shell: |
+      kubectl delete machine "{{ WORKER_NODE }}" -n "{{ NAMESPACE }}"
+
+  - name: Wait until one BMH is in ready state
```
[the remark this answers is not in the document]
Comment (smoshiur1237) [reply vm-setup/roles/v1aX_integration_test/tasks/remediation.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml [hunk 5]
```diff
@@ -138,4 +138,83 @@
...
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | wc -l
+    retries: 200
+    delay: 20
+    register: ready_hosts
+    until: ready_hosts.stdout == "1"
```
[the remark this answers is not in the document]
Comment (smoshiur1237) [reply vm-setup/roles/v1aX_integration_test/tasks/remediation.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml [hunk 6]
```diff
@@ -138,4 +138,83 @@
...
+
+  - name: Get machinedeployment object name
+    shell: |
+      kubectl get machinedeployment -n metal3 -o json | jq -r '.items[]
+      | .metadata.name'
+    register: machinedeployment_name
+
+  - name: Define machine deployment
+    set_fact:
+      MACHINE_DEPLOYMENT_NAME: "{{ machinedeployment_name.stdout }}"
```
[the remark this answers is not in the document]
Comment (smoshiur1237) [reply vm-setup/roles/v1aX_integration_test/tasks/remediation.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml (hunk 2 above)
Comment (smoshiur1237) [reply vm-setup/roles/v1aX_integration_test/tasks/remediation.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml (hunk 1 above)
Comment (smoshiur1237) [reply vm-setup/roles/v1aX_integration_test/tasks/remediation.yml]: Done

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml [hunk 7]
```diff
@@ -12,7 +12,19 @@
...
+
+  - name: Get KCP name
+    shell: |
+      kubectl get kcp -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
+      | .metadata.name'
+    register: kcp_name
+
+  - name: Get machinedeployment object name
+    shell: |
+      kubectl get machinedeployment -n metal3 -o json | jq -r '.items[]
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:24]: Please parameterize the namespace here.

File: vm-setup/roles/v1aX_integration_test/tasks/remediation.yml [hunk 8]
```diff
@@ -138,4 +152,64 @@
...
+    register: provisioned_hosts
+    until: provisioned_hosts.stdout == "3"
+
+  - name: Scale up the machinedeployment
+    shell: kubectl scale machinedeployment "{{ MACHINE_DEPLOYMENT_NAME }}" -n "{{ NAMESPACE }}" --replicas=2
+  
+  - pause:
+     minutes: 1
+
+  - name: Wait and verify that none of the nodes will not start provisioning
```
Comment (fmuyassarov) [inline vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:197]: Typo in the name of the task.

**Review by fmuyassarov — COMMENTED** [review]
Minor comments, otherwise looks good.

**Pull request merged**